observant-otters/forumthing/forum/models.py

```diff
@@ -1,3 +1,31 @@
 from django.db import models
+from user.models import User
 
-# Create your models here.
+class Thread(models.Model):
+    """
+    the model for a thread/conversation
+    """
+    title = models.TextField()
+    date_created = models.DateTimeField()
+    author = models.OneToOneField(User, models.CASCADE)
+
+
+class Message(models.Model):
+    """
+    the model for a message inside a thread
+    """
+    content = models.TextField()
+    thread = models.ForeignKey(Thread, on_delete=models.CASCADE)
+    author = models.OneToOneField(User, models.DO_NOTHING)  # it's either this or models.PROTECT
+    date_posted = models.DateTimeField()
+    date_edited = models.DateTimeField(default=date_posted)
+
+
+class UserThreadEvent(models.Model):
+    """
+    we log an event when the user interacts with the thread,
+    whether they be viewing it or making a post
+    """
+    type = models.BooleanField()  # idk, true would be create, and false would be view?
```

Comment:
Let's make this a string, in case we ever want to add new kinds of events.

Or better, does Django support an enum field type?

Reply:
Well, I found this- (the enum support thread I found on SO was for 2.0 and lower): [https://github.com/2degrees/django-enum-field](url)